lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,416 @@
+//! UI component that allows picking a number through mouse interaction. Horizontal mouse movement
+//! changes the value within a set range. Vertical mouse movement changes the precision of the
+//! slider, this influences the response to horizontal mouse movement and the increment steps to
+//! which the value is snapped.
+
+#![recursion_limit = "1024"]
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+#![allow(clippy::bool_to_int_with_if)]
+#![allow(clippy::let_and_return)]
+
+use crate::model::*;
+use ensogl_core::prelude::*;
+
+use ensogl_core::application;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::Animation;
+use ensogl_text::formatting;
+
+
+// ==============
+// === Export ===
+// ==============
+
+pub mod model;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// Base of the exponentiation for precision increment steps of 10^x
+pub const PRECISION_STEP_BASE: f32 = 10.0;
+
+
+
+// ========================
+// === Slider component ===
+// ========================
+
+/// Slider component structure
+pub struct Slider {
+    pub frp: Frp,
+    model:   Rc<Model>,
+    pub app: Application,
+}
+
+impl Slider {
+    /// Constructor
+    pub fn new(app: &Application) -> Self {
+        let model = Rc::new(Model::new(app));
+        let app = app.clone_ref();
+        let frp = Frp::new();
+
+        Self { frp, model, app }.init()
+    }
+}
+
+impl display::Object for Slider {
+    fn display_object(&self) -> &display::object::Instance {
+        self.model.display_object()
+    }
+}
+
+impl Deref for Slider {
+    type Target = Frp;
+    fn deref(&self) -> &Self::Target {
+        &self.frp
+    }
+}
+
+impl FrpNetworkProvider for Slider {
+    fn network(&self) -> &enso_frp::Network {
+        self.frp.network()
+    }
+}
+
+impl application::View for Slider {
+    fn label() -> &'static str {
+        "Slider"
+    }
+
+    fn new(app: &Application) -> Self {
+        Self::new(app)
+    }
+
+    fn app(&self) -> &Application {
+        &self.app
+    }
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+ensogl_core::define_endpoints_2! {
+    Input {
+        /// Set the width of the component
+        set_width(f32),
+        /// Set the height of the component
+        set_height(f32),
+
+        /// Set the color of the slider
+        set_slider_color(color::Lcha),
```

Comment:
The whole component is a slider, so what does it mean to set the slider color? Is it setting the color of the whole component?